eth/vm/gas_meter.py

```diff
@@ -16,17 +17,49 @@
 )
 
 
+def default_refund_strategy(meter: "GasMeter", amount: int) -> None:
+    if amount < 0:
+        raise ValidationError("Gas refund amount must be positive")
+
+    meter.gas_refunded += amount
+
+    meter.logger.trace(
+        'GAS REFUND: %s + %s -> %s',
+        meter.gas_refunded - amount,
+        amount,
+        meter.gas_refunded,
+    )
+
+
+def allow_negative_refund_strategy(meter: "GasMeter", amount: int) -> None:
+    meter.gas_refunded += amount
+
+    meter.logger.trace(
+        'GAS REFUND: %s + %s -> %s',
+        meter.gas_refunded - amount,
+        amount,
+        meter.gas_refunded,
+    )
```

Comment:
Looks like these logs can stay in the `GasMeter`

Reply:
I was wondering about that and then said to myself: If it's really just this single line, why not duplicate it to allow *entirely* swapping out the implementation including the generated log.